test/sort-rules.test.ts

```diff
@@ -18,19 +25,24 @@ describe('sort rules', () => {
     expect(await sort('pt-2 p-4 foo'))
       .toMatchInlineSnapshot('"foo p-4 pt-2"')
     expect(await sort('hover:focus:p1 hover:mt1 hover:m2 pt-2 p-4'))
-      .toMatchInlineSnapshot('"p-4 pt-2 hover:m2 hover:mt1 hover:focus:p1"')
+      .toMatchInlineSnapshot('"hover:focus:p1 hover:m2 hover:mt1 p-4 pt-2"')
```

Comment:
These do not look ideal me. Variants should be after normal rules.

Reply:
If we sort the codes corresponding to the variants in the preset, will it be more difficult for us to judge how to write class in order?
```javascript
// sort-rules.ts
let sorted = result
    .filter(notNull)
    .sort((a, b) => {
      let result = a[0] - b[0]
      if (result === 0)
        result = a[1].localeCompare(b[1])
      return result
    })
    .map(i => i[1])
    .join(' ')
```
when I set up `shortCuts`, I don't know what the code corresponding to my class is. As a result, I cannot actively modify the sorting, so is it better to sort alphabetically?

Reply:
I think the goal of the eslint plugin is to sort the rules by their definition order, so rules with similar scopes are placed together. Otherwise, if we sort alphabetically, you might not need this config to be UnoCSS-specific.

To tackle this, I think we should have a higher sort by the CSS layers, so shortcuts are separate from utilities classes. In the shortcuts layer, we might be able to sort alphabetically.

Reply:
> I think the goal of the eslint plugin is to sort the rules by their definition order, so rules with similar scopes are placed together. Otherwise, if we sort alphabetically, you might not need this config to be UnoCSS-specific.
> 
> To tackle this, I think we should have a higher sort by the CSS layers, so shortcuts are separate from utilities classes. In the shortcuts layer, we might be able to sort alphabetically.

thank you for your reply. I may not understand what you said. Do you mean we need to pay attention to the writing order in short cuts ?